internal/engine/k8swatch/pod_watch.go

```diff
@@ -152,6 +152,37 @@ func (w *PodWatcher) removeWatch(toRemove PodWatch) {
 	}
 }
 
+// We inject a label to all pod template specs so that we can tell if it came from a pod template we've deployed
+// since startup.
+// This is to solve the problem where:
+// 1. start Tilt, which applies a deployment and creates a pod
+// 2. kill Tilt, change some code that goes into the pod's image
+// 3. start Tilt, which reapplies that same deployment
+// At this point, we want a new pod to be created, and we don't care about the old pod. However, since
+// changing a deployment's PodTemplateSpec does not cause the deployment to get a new UID, the deployment
+// applied in (3) has the same UID as the deployment applied in (1), so when we get an event for the
+// undesired pod from (1), we trace it back to the deployment we applied in (3), and emit an event.
+// To deal with this, we inject a hash of the pod template spec into the pod template itself, and only
+// emit events from pods with known pod template spec hashes.
+func isKnownPodTemplateSpecHash(pod *v1.Pod, manifest *store.ManifestTarget) bool {
+	// if it matches ExtraPodSelectors, don't worry about the hash - for CRDs using ExtraPodSelectors
+	// there isn't necessarily a PodTemplateSpec into which we can put a hash
+	for _, selector := range manifest.Manifest.K8sTarget().ExtraPodSelectors {
+		if selector.Matches(labels.Set(pod.Labels)) {
+			return true
+		}
+	}
+
+	ms := manifest.State
+	if !ms.IsK8s() {
+		return false
+	}
+
+	hash, ok := pod.Labels[k8s.TiltPodTemplateHashLabel]
```

Comment:
i'm pretty sure we want to keep pods that don't have a podTemplate hash labels? otherwise we will break people who are using this with resources that don't have a PodTemplate

Reply:
Hm, my mental model was that all pods are either coming from "normal" things, and have specs, or from CRDs, and have ExtraPodLabels. I hadn't put a lot of thought into that, though. And now that you've made me think about it, I guess it's possible that the UID stuff might solve the CRD/ExtraPodLabel issue without having to specify labels?

I was slightly concerned that we'd risk picking up a pod that was deployed by non/old tilt and didn't have a label, but that doesn't really seem worth worrying about.